api/src/opentrons/protocol_engine/state/tips.py

```diff
@@ -73,6 +81,28 @@ def handle_action(self, action: Action) -> None:
                     well_name
                 ] = TipRackWellState.CLEAN
 
+        elif isinstance(action, AddPipetteConfigAction):
+            self._state.channels_by_pipette_id[action.pipette_id] = action.channels
+
+    def _set_used_tips(self, pipette_id: str, well_name: str, labware_id: str) -> None:
+        pipette_channels = self._state.channels_by_pipette_id.get(pipette_id)
+        columns = self._state.column_by_labware_id[labware_id]
+        tips_in_labware = self._state.tips_by_labware_id[labware_id]
+
+        if pipette_channels == len(tips_in_labware.keys()):
```

Comment:
I think the `keys` here is not necessary

```suggestion
        if pipette_channels == len(tips_in_labware):
```